src/eins-network-id.c

```diff
@@ -0,0 +1,238 @@
+/* Copyright 2017 Endless Mobile, Inc.
+ *
+ * This file is part of eos-metrics-instrumentation.
+ *
+ * eos-metrics-instrumentation is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * eos-metrics-instrumentation is distributed in the hope that it will be
+ * useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with eos-metrics-instrumentation.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+#include <glib.h>
+#include <stdio.h>
+
+#include "eins-network-id.h"
+
+#define PROC_IPV4_ROUTE_FILE "/proc/net/route"
+#define IPV4_ROUTE_REGEXP "^(?:\\S+\\s+){2}(?P<Gateway>[0-9A-F]{8})\\s+(?:\\S+\\s+){4}(?P<Mask>00000000)"
```

Comment:
Do you also want to allow `a-f` (lowercase) for the `<Gateway>`? This might not be the format which appears in the kernel at the moment, but it seems wise to be lenient with the parsing here.